spacy/tokens/span.pyx

```diff
@@ -639,7 +639,7 @@ cdef class Span:
         else:
             return self.doc[root]
 
-    def char_span(self, int start_idx, int end_idx, label=0, kb_id=0, vector=None, id=0):
+    def char_span(self, int start_idx, int end_idx, label=0, kb_id=0, vector=None, alignment_mode="strict", id=0):
```

Comment:
I see that you were trying to match the order in `Doc.char_span`, but if anyone has been using positional args here, adding a new arg in the middle can break existing code:

```suggestion
    def char_span(self, int start_idx, int end_idx, label=0, kb_id=0, vector=None, id=0, alignment_mode="strict"):
```

This result is frustrating for the `Doc` vs. `Span` API (well, for people using positional args), but we can put it on our list of things to clean up in the API for v4.0.

Reply:
The `id` parameter is undocumented and not used in the function body. Wonder why it's there to start with. Still wanna put it before `alignment_mode`?

Reply:
`id` should be passed as `span_id` but isn't. I would still append `alignment_mode`. It's not a huge issue and we can straighten it out for v4.